src/Exception/InvalidArgumentException.php

```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace Tnapf\JsonMapper\Exception;
+
+use Tnapf\JsonMapper\MapperException;
+
+class InvalidArgumentException extends MapperException
+{
```

Comment:
Not sure how I feel about having exceptions dedicated for this. I presume this is just so catching is more precise?

Reply:
> I presume this is just so catching is more precise?

Yes, precisely. This should've been a runtime exception, but then it would require extracting `MapperExceptionInterface` and I didn't feel like doing so.